dali/operators/decoder/audio/audio_decoder_op.cc

```diff
@@ -62,7 +62,7 @@ AudioDecoderCpu::SetupImpl(std::vector<OutputDesc> &output_desc, const workspace
   sample_meta_.resize(batch_size);
   files_names_.resize(batch_size);
 
-  decode_type_ = (use_resampling_ && !downmix_) ? DALI_FLOAT : output_type_;
```

Comment:
downmix_ is not enough. If the number of channels was already 1 there won't be downmixing